### PR TITLE
sorting defaults to to:desc and generic try catch for servlet

### PR DESCRIPTION
**Pull request by KirbyKatcher**

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java [hunk 1]
```diff
@@ -67,7 +68,16 @@ public class RunQuery extends HttpServlet {
 
 	@Override
 	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		try{
+			retrieveResults(req, resp);
+		}catch (Exception e){
+			String msg = new ServletError(GAL5000_GENERIC_API_ERROR).toString();
+					sendResponse(resp, msg, HttpServletResponse.SC_NOT_FOUND);
```
Comment (hobbit1983) [inline galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:75]: Is a 404 the best response when handling a generic error like this?

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/ServletErrorMessage.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 public enum ServletErrorMessage {
 
+    GAL5000_GENERIC_API_ERROR                       (5000,"E: Error occured when trying to access the endpoint. Report the problem to your Galasa server owner."),
```
Comment (hobbit1983) [inline galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/ServletErrorMessage.java:8]: would this be better to say Galasa EcoSystem owner?

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java [hunk 3]
```diff
@@ -67,7 +68,16 @@ public class RunQuery extends HttpServlet {
...
+					logger.error(msg,e);
+		}
+	};
 
+	protected void retrieveResults(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
 		int pageNum = 1;
 		int pageSize = 100;
```
Comment (hobbit1983) [inline galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:82]: I know this isn't part of your PR - but why define these values here when we give them new values in 2 lines time?  Can we clean this up a bit?

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java [hunk 4]
```diff
@@ -269,20 +279,20 @@ private JsonObject pageToJson(List<RasRunResult> resultsInPage, int totalRuns, i
...
 
+		// Collect all the runs from all the RAS stores into a single list
+		List<IRunResult> runs = new ArrayList<>();
 		for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
-
 			runs.addAll(directoryService.getRuns(criteria));
-
 		}
 
+		// Convert each result to the required format and
```
Comment (hobbit1983) [inline galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:294]: incomplete comment?

**Review by hobbit1983 — CHANGES_REQUESTED** [review]

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java (hunk 3 above)
Comment (techcobweb) [reply galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java]: Fixed in latest code.

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java (hunk 1 above)
Comment (techcobweb) [reply galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java]: Fixed today with @AaronJhaj

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by AaronJhaj — APPROVED** [review]

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by hobbit1983 — CHANGES_REQUESTED** [review]

**Comment by galasa-team** [comment]
Build successful

**Review by hobbit1983 — APPROVED** [review]

**Pull request merged**